File: backend/app/schemas/study.py

from pydantic import BaseModel
from typing import List, Optional, Dict
from uuid import UUID
import datetime

class QuestionBase(BaseModel):
    text: str
    type: Optional[str] = None

class QuestionCreate(QuestionBase):
    pass

class Question(QuestionBase):
    id: UUID
    study_id: UUID

    class Config:
        orm_mode = True

class StudyBase(BaseModel):
    title: str
    description: Optional[str] = None
    methodology: Optional[str] = None

class StudyCreate(StudyBase):
    questions: List[QuestionCreate] = []
    unique_code: Optional[str] = None

class Study(StudyBase):
    id: UUID
    created_at: datetime.datetime
    unique_code: str
    questions: List[Question] = []

    class Config:
        orm_mode = True

class AnswerBase(BaseModel):
    text: str

class AnswerCreate(AnswerBase):
    question_id: UUID
    user_id: UUID

class Answer(AnswerBase):
    id: UUID
    study_id: UUID
    question_id: UUID
    user_id: UUID

    class Config:
        orm_mode = True

class AnswerSubmission(BaseModel):
    user_id: UUID
    answers: Dict[str, str]

File: client/src/types/study.ts
export type QuestionType = "qualitative" | "quantitative";
export type Methodology = "Survey" | "KII";

export interface Question {
  id: string;
  text: string;
  type: QuestionType;
  study_id: string;
}

export interface QuestionInput {
  text: string;
  type: QuestionType;
}

export interface Study {
  id: string;
  title: string;
  description: string;
  methodology: Methodology;
  questions: Question[];
  unique_code: string;
  author_id: string;
  created_at: string;
}

export interface StudyCreateInput {
  title: string;
  description: string;
  methodology: Methodology;
  questions: QuestionInput[];
}

export interface AnswerSubmission {
  user_id: string;
  answers: Record<string, string>; // question_id -> answer_text
}

export interface Quote {
  text: string;
  author: string;
}

export interface Theme {
  keyword: string;
  count: number;
}

export interface SentimentBreakdown {
  positive: number;
  negative: number;
  neutral: number;
}

export interface AnalysisResult {
  study_id: string;
  participant_count: number;
  sentiment: SentimentBreakdown;
  themes: Theme[];
  key_quotes: Quote[];
  insights: string[];
  answers_table: {
    headers: string[];
    rows: string[][];
  };
}

export type StudyUserMode = "creator" | "participant";
export type CreatorTab = "build" | "dashboard" | "findings" | "methodology";
export type ParticipantStep = "gate" | "consent" | "questionnaire" | "complete";
